probe/plugins/unix_round_tripper.go

```diff
@@ -0,0 +1,27 @@
+package plugins
+
+import (
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"time"
+)
+
+type unixRoundTripper struct {
+	address string
+	timeout time.Duration
+}
+
+func makeUnixRoundTripper(address string, timeout time.Duration) (http.RoundTripper, error) {
+	return unixRoundTripper{address: address, timeout: timeout}, nil
+}
+
+func (t unixRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	conn, err := net.DialTimeout("unix", t.address, t.timeout)
+	if err != nil {
+		return nil, err
+	}
+	client := httputil.NewClientConn(conn, nil)
```

Comment:
I hadn't read this carefully. I think we could simply reuse the connection here, which would be more efficient and would let you detect removals of the unix socket without races.